src/Functions/MatchImpl.h

```diff
@@ -118,6 +118,19 @@ struct MatchImpl
         if (haystack_offsets.empty())
             return;
 
+        /// Fast path for [I]LIKE, because the result is always true or false
```

Comment:
Besides in `vectorConstant()`, we should have the same optimization also in `vectorFixedConstant()` (which is called if the inputs are of type `FixedString`).

There are two methods `vector*vector()` which get called for non-const haystacks. This case is slow and unoptimized anyways and accordingly it misses a bunch of optimizations. No need to add the shortcut to these two methods.